tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/compute.py

```diff
@@ -579,6 +587,20 @@ def _operation_status_done(operation):
     def _execute(  # pylint: disable=arguments-differ
         self, request, *, timeout_sec=_WAIT_FOR_OPERATION_SEC
     ):
+        if xds_flags.ENABLE_GFE_DEBUG_HEADER.value:
+            old_postproc = request.postproc
+
+            def _log_debug_header(resp, contents):
+                if DEBUG_HEADER_IN_RESPONSE in resp:
+                    logger.info(
+                        "Received debug headers: %s",
+                        resp[DEBUG_HEADER_IN_RESPONSE],
+                    )
+                return old_postproc(resp, contents)
+
+            logger.info("Adding debug headers for method: %s", request.methodId)
+            request.headers[DEBUG_HEADER_KEY] = DEBUG_HEADER_VALUE
+            request.postproc = _log_debug_header
```

Comment:
Looks like we need `response_callbacks` (see `add_response_callback`) instead of overriding `request.postproc`. It's called just before `postproc`, but not after raising on an HTTP error (which we actually want):

```py
        for callback in self.response_callbacks:
            callback(resp)
        if resp.status >= 300:
            raise HttpError(resp, content, uri=self.uri)
        return self.postproc(resp, content)
```

https://github.com/googleapis/google-api-python-client/blob/d1bd28f839c8da94d043606ed0b66ab60ff27545/googleapiclient/http.py#L912-L916

Reply:
Right, that's a better place to put our callback, will change it.